runtime/ftplugin/vim.vim

```diff
@@ -48,7 +49,49 @@ setlocal fo-=t fo+=croql
 setlocal isk+=#
 
 " Use :help to lookup the keyword under the cursor with K.
-setlocal keywordprg=:help
+" Distinguish between commands, options and functions.
+if !exists("*" .. expand("<SID>") .. "Help")
+  function s:Help(topic) abort
+    let topic = a:topic
+
+    if get(g:, 'syntax_on', 0)
+      silent! let syn = synIDattr(synID(line('.'), col('.'), 1), 'name')
```

Comment:
why the silent!?

Reply:
and why would this fail when syntax highlighting is enabled?

Reply:
It's just a precautionary measure to exclude unpredicted failures